File: app/src/main/java/com/bq/zowi/controllers/RankingController.java
package com.bq.zowi.controllers;

import com.bq.zowi.controllers.GameController;
import com.bq.zowi.models.RankingEntry;
import java.util.ArrayList;
import rx.Single;

/* JADX INFO: loaded from: classes.dex */
public interface RankingController {
    Single<ArrayList<RankingEntry>> getRanking(GameController.GAME_ID game_id);

    Single<Boolean> isScoreInTop10(GameController.GAME_ID game_id, int i);

    Single<Void> resetAllRankings();

    Single<ArrayList<RankingEntry>> saveRankingEntry(GameController.GAME_ID game_id, RankingEntry rankingEntry);
}

File: zowi-core/src/main/kotlin/com/bq/zowi/api/RankingController.kt
package com.bq.zowi.api

import com.bq.zowi.models.RankingEntry
import io.reactivex.Single

interface RankingController {
    fun getRanking(gameId: GameController.GAME_ID): Single<ArrayList<RankingEntry>>
    fun isScoreInTop10(gameId: GameController.GAME_ID, score: Int): Single<Boolean>
    fun resetAllRankings(): Single<Void>
    fun saveRankingEntry(gameId: GameController.GAME_ID, entry: RankingEntry): Single<ArrayList<RankingEntry>>
}
